src/components/button/button.test.tsx

```diff
@@ -1,12 +1,51 @@
 import React from 'react';
-import { render, screen } from '@testing-library/react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom';
 
 import Button from './button';
+import { ButtonProps } from './button';
+
+function renderAndGetButton(props: ButtonProps): HTMLElement {
+  render(<Button label={props.label || 'Button'} {...props} />);
+  return screen.getByRole('button', { name: props.label });
+}
 
 describe('Button', () => {
   test('renders the Button component', () => {
     const label = 'Hello world!';
-    render(<Button label={label} />);
+    render(<Button type="primary" label={label} />);
     expect(screen.queryAllByText(label)).toHaveLength(1);
   });
+
+  test('execute the onClick event', () => {
```

Comment:
```suggestion
  test('execute the onClick event', () => {
```

Prefer start spec with `should`